src/librustdoc/json/conversions.rs

```diff
@@ -162,8 +162,8 @@ impl From<clean::ItemKind> for ItemEnum {
             ForeignFunctionItem(f) => ItemEnum::FunctionItem(f.into()),
             TraitItem(t) => ItemEnum::TraitItem(t.into()),
             TraitAliasItem(t) => ItemEnum::TraitAliasItem(t.into()),
-            MethodItem(m, _) => ItemEnum::MethodItem(m.into()),
-            TyMethodItem(m) => ItemEnum::MethodItem(m.into()),
+            MethodItem(m, _) => ItemEnum::MethodItem(from_function_method(m, true)),
+            TyMethodItem(m) => ItemEnum::MethodItem(from_function_method(m, false)),
```

Comment:
This doesn't look right. You can have default implementations of an associated function:
```rust
trait Trait {
  fn foo() {}
}
```

Reply:
Oh wait, you added a test case for that. I'm confused then - does rustdoc use `MethodItem` even for functions in the trait itself?

Reply:
`TyMethodItem` is for signatures, `MethodItem` is for any method with a body, in traits or not. At least, that's what the documentation says, and usage seems to agree